src/core_plugins/i18n/server/i18n/i18n.js

```diff
@@ -0,0 +1,128 @@
+import { readFile } from 'fs';
+import path from 'path';
+import Promise from 'bluebird';
+import _ from 'lodash';
+
+const asyncReadFile = Promise.promisify(readFile);
+
+const TRANSLATION_FILE_EXTENSION = '.json';
+
+let registeredTranslations = {};
+
+/**
+ * Return all translations registered for a particular locale.
+ * @param {string} locale - Translation locale to be returned
+ * @return {Promise} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+const getTranslationsForLocale = function (locale) {
+  if (!registeredTranslations.hasOwnProperty(locale)) {
+    return Promise.resolve(null);
+  }
+
+  const translationFiles = registeredTranslations[locale];
+  const translations = _.map(translationFiles, (filename) => {
+    return asyncReadFile(filename, 'utf8').then(fileContents => JSON.parse(fileContents)).catch(SyntaxError, function (e) {
+      return Promise.reject('Invalid json in ' + filename);
+    }).catch(function (e) {
+      return Promise.reject('Cannot read file ' + filename);
+    });
+  });
+
+  return Promise.all(translations).then(translations => _.assign({}, ...translations)).then(function (translations) {
+    const vals = Object.values(translations);
+    if (!vals.every(isString)) return  Promise.reject('Invalid json schema for translations');
+    return translations;
+  });
+};
+
+/**
+ * Returns list of all registered locales.
+ * @return {Array} - Registered locales
+ */
+const getRegisteredTranslationLocales = function () {
+  return Object.keys(registeredTranslations);
+};
+
+/**
+ * Return translations for a suitable locale from a user side locale list
+ * @param {array} acceptLanguages - Array of locale objects parsedfrom user side accept languages header
+ * @param {string} defaultLocale - Default locale as configured in Kibana
+ * @return {object} - A Promise object which will contain on resolve a JSON object of all registered translations
+ */
+const getTranslationsForLocales = function (acceptLanguages, defaultLocale) {
```

Comment:
Also, I think the i18n module should already know what the default locale is. It could be set at plugin initialization time. But this particular method probably shouldn't fallback to the default at all. There should be a separate `getDefaultTranslations` method that the client can use if this method returns empty results.